### ISSUE TITLE
Investigate interaction of robust reweighting, uv-cut and writing of corrected weight.

### ISSUE THREAD

**Issue by JSKenyon** [ratt-ru/QuartiCal#283]
**Describe the bug**
Writing out the corrected weight/weight when using a uv-cut may result in zero weight on baselines deselected by the cut. This is both a bug and a sharp edge although it is connected to relatively specialised functionality. I had begun fixing this in #225 but I never came to a definitive conclusion about the correct approach. The problem is also further exacerbated by the fact that QuartiCal currently only allows for one instance of the weight column (for memory reasons). However, this is problematic in cases where the robust solver is in use as the user may want to use the robust weights during calibration but write out the corrected weights based on the current contents of WEIGHT_SPECTRUM. This is necessary as imaging with the robust weights can (will) suppress unmodelled flux. 

**To Reproduce**
Run QC with the robust solver and write out corrected_weight.

**Expected behavior**
QC should:

1. Treat the UV-cut correctly during reweighting.
2. Support writing out corrected weight using contents of WEIGHT_SPECTRUM even when robust reweighting is enabled.

**Workaround**
Running QC again with `solver.robust=False` and loading the previously solved gains will allow writing corrected weight from either SIGMA_SPECTRUM/WEIGHT_SPECTRUM. This is obviously tedious but should work in the interim.

**Comment by ktrehaeven** [comment]
Hi @JSKenyon would I also be able to peel during the first and/or second run of QC?

**Comment by JSKenyon** [comment]
> Hi @JSKenyon would I also be able to peel during the first and/or second run of QC?

In general? There is nothing stopping you from peeling from the outset if you have the relevant model components. Not sure if that answers your question though.

**Comment by ktrehaeven** [comment]
Yup. And when doing the second run, should the input column be the CORRECTED_DATA outputted from the first run or just DATA?
